src/main/java/org/opensearch/neuralsearch/util/PluginFeatureFlags.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.util;
+
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.core.common.Strings;
+import org.opensearch.transport.TransportSettings;
+
+/**
+ * Abstracts feature flags operations specific to neural-search plugin
+ */
+public class PluginFeatureFlags {
+
+    /**
+     * Used to test feature flags whose values are expected to be booleans.
+     * This method returns true if the value is "true" (case-insensitive),
+     * and false otherwise.
+     * Checks alternative flag names as they may be different for plugins
+     */
+    public static boolean isEnabled(String featureFlagName) {
```

Comment:
we need to try "transport" feature prefix, as this is the way core adding features that are set in plugins.

Reply:
Then why both? Should we only check with "transport" feature prefix?

Reply:
I wanted to encapsulate that logic so client can do only one call. Otherwise on a client we need to do two calls:
```featureFlags.isEnabled(flag) || customFeatureFlags.isEnabled(flag)```. Depending on scenario one feature flag can be set in one form or another (with or without prefix)

Reply:
I still think it is client's responsibility to pass correct key. In this case `transport.features.neural_search_hybrid_search_enabled`. 

There is no such case that `neural_search_hybrid_search_enabled` can be enabled by setting `neural_search_hybrid_search_enabled` as true but only by setting `transport.features.neural_search_hybrid_search_enabled` as true I believe?

Reply:
I see your point, it can be that flag is set in two places and as we read it with OR and one that set in plugin is always TRUE then we cannot set it to FALSE ever. I'll rework to only use one name or another, in fact it should be always with the prefix except the definition.